In [1]:
import pandas as pd
import datetime as dt

In [2]:
LCK = pd.read_csv('projectData/LCK_RESULT.csv', index_col=0)
LCK['version'] = LCK['version'].str.replace('버전','')
LCK['version'] = LCK['version'].str.replace(' ','')
LCK.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1215 entries, 0 to 1214
Data columns (total 29 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   game_date          1215 non-null   object
 1   version            1215 non-null   object
 2   team1_name         1215 non-null   object
 3   team1_score        1215 non-null   int64 
 4   team2_score        1215 non-null   int64 
 5   team2_name         1215 non-null   object
 6   gametime           1215 non-null   object
 7   blue_team          1215 non-null   object
 8   blue_score         1215 non-null   int64 
 9   blue_ban           1215 non-null   object
 10  blue_object        1215 non-null   object
 11  blue_object_count  1215 non-null   int64 
 12  blue_call_name     1215 non-null   object
 13  blue_line          1215 non-null   object
 14  blue_champion      1215 non-null   object
 15  blue_kill          1215 non-null   int64 
 16  blue_death         1215 non-null   object


In [83]:
LCK

game_date version team1_name  team1_score  team2_score team2_name  \
0     2021.09.02 (목)   11.16         T1            3            2        HLE   
1     2021.09.02 (목)   11.16         T1            3            2        HLE   
2     2021.09.02 (목)   11.16         T1            3            2        HLE   
3     2021.09.02 (목)   11.16         T1            3            2        HLE   
4     2021.09.02 (목)   11.16         T1            3            2        HLE   
...              ...     ...        ...          ...          ...        ...   
1210  2021.06.11 (금)   11.11         DK            2            1         T1   
1211  2021.06.11 (금)   11.11         DK            2            1         T1   
1212  2021.06.11 (금)   11.11         DK            2            1         T1   
1213  2021.06.11 (금)   11.11         DK            2            1         T1   
1214  2021.06.11 (금)   11.11         DK            2            1         T1   

     gametime blue_team  blue_score   blue_ban  ... red_score  red_ban  \
0       36:48        T1          20        레넥톤  ...         4      라이즈   
1       36:48        T1          20         카밀  ...         4      제이스   
2       36:48        T1          20        르블랑  ...         4      비에고   
3       36:48        T1          20        레오나  ...         4     칼리스타   
4       36:48        T1          20        브라움  ...         4       애쉬   
...       ...       ...         ...        ...  ...       ...      ...   
1210    37:11        DK          15         그웬  ...        13      레넥톤   
1211    37:11        DK          15       이즈리얼  ...        13       세나   
1212    37:11        DK          15        리 신  ...        13      루시안   
1213    37:11        DK          15  트위스티드 페이트  ...        13       룰루   
1214    37:11        DK          15       그라가스  ...        13       조이   

          red_object red_object_count red_call_name  red_line red_champion  \
0         turret@red                1    HLE Morgan       TOP           나르   
1      inhibitor@red                0    HLE Willer        JG         신 짜오   
2          baron@red                0     HLE Chovy       MID          아지르   
3         dragon@red                2      HLE Deft        AD          바루스   
4     riftHerald@red                0      HLE Vsta       SUP          카르마   
...              ...              ...           ...       ...          ...   
1210      turret@red                2      T1 Canna       TOP           세트   
1211   inhibitor@red                0       T1 Cuzz        JG           럼블   
1212       baron@red                0      T1 Faker       MID          제이스   
1213      dragon@red                2      T1 Teddy        AD         칼리스타   
1214  riftHerald@red                0      T1 Keria       SUP          레오나   

     red_kill red_death  red_assist  
0           1         6           1  
1           1         4           2  
2           1         3           3  
3           1         4           3  
4           0         3           4  
...       ...       ...         ...  
1210        1         2           4  
1211        3         3           5  
1212        2         3           3  
1213        7         4           4  
1214        0         3           9  

[1215 rows x 29 columns]

In [84]:
ban_a  =LCK[['blue_ban']]
ban_a.columns=['ban']
ban_b  =LCK[['red_ban']]
ban_b.columns=['ban']
ban_all =pd.concat([ban_a,ban_b])

In [85]:
ban_all.value_counts()

ban 
루시안     152
레넥톤     142
칼리스타    109
제이스     100
신 짜오     97
       ... 
빅토르       1
타릭        1
소라카       1
사미라       1
사이온       1
Length: 76, dtype: int64

In [113]:
LCK_ver11 = LCK[LCK['version'] == '11.11']
LCK_ver12 = LCK[LCK['version'] == '11.12']
LCK_ver13 = LCK[LCK['version'] == '11.13']
LCK_ver14 = LCK[LCK['version'] == '11.14']
LCK_ver15 = LCK[LCK['version'] == '11.15']

In [110]:
LCK_ver11.columns

Index(['game_date', 'version', 'team1_name', 'team1_score', 'team2_score',
       'team2_name', 'gametime', 'blue_team', 'blue_score', 'blue_ban',
       'blue_object', 'blue_object_count', 'blue_call_name', 'blue_line',
       'blue_champion', 'blue_kill', 'blue_death', 'blue_assist', 'red_team',
       'red_score', 'red_ban', 'red_object', 'red_object_count',
       'red_call_name', 'red_line', 'red_champion', 'red_kill', 'red_death',
       'red_assist'],
      dtype='object')

## 확인할거 평균 게임시간
## 오브젝트 관련해서 어떤걸 많이 먹는지
## 나는 버전별로 구분해서 확인하겠음

In [120]:

# v11.11
cham_a_11 = LCK_ver11[['blue_champion']]
cham_a_11.columns=['champion']

cham_b_11 = LCK_ver11[['red_champion']]
cham_b_11.columns=['champion']
cahm_all_v11 = pd.concat([cham_a_11, cham_b_11])


# v11.12
cham_a_12 = LCK_ver12[['blue_champion']]
cham_a_12.columns=['champion']

cham_b_12 = LCK_ver12[['red_champion']]
cham_b_12.columns=['champion']
cahm_all_v12 = pd.concat([cham_a_12, cham_b_12])

# v11.13
cham_a_13 = LCK_ver13[['blue_champion']]
cham_a_13.columns=['champion']

cham_b_13 = LCK_ver13[['red_champion']]
cham_b_13.columns=['champion']
cahm_all_v13 = pd.concat([cham_a_13, cham_b_13])


# v11.14
cham_a_14 = LCK_ver14[['blue_champion']]
cham_a_14.columns=['champion']

cham_b_14 = LCK_ver14[['red_champion']]
cham_b_14.columns=['champion']
cahm_all_v14 = pd.concat([cham_a_14, cham_b_14])

# v11.15
cham_a_15 = LCK_ver15[['blue_champion']]
cham_a_15.columns=['champion']

cham_b_15 = LCK_ver15[['red_champion']]
cham_b_15.columns=['champion']
cahm_all_v15 = pd.concat([cham_a_15, cham_b_15])

In [127]:
cahm_all_v11.value_counts().head(20)

champion
럼블          22
세트          21
이즈리얼        19
리 신         19
우디르         16
바루스         16
녹턴          16
다이애나        16
나르          12
탐 켄치        12
레오나         11
아칼리         10
카르마         10
칼리스타         9
비에고          9
오리아나         9
그웬           9
징크스          9
쓰레쉬          8
카이사          8
dtype: int64

In [123]:
cahm_all_v12.value_counts().head(10)

champion
럼블          35
세트          32
리 신         26
다이애나        24
비에고         23
레오나         20
칼리스타        19
레넥톤         18
바루스         17
쓰레쉬         16
dtype: int64

In [124]:
cahm_all_v13.value_counts().head(10)

champion
리 신         29
레오나         27
라이즈         21
그웬          21
바루스         20
다이애나        20
레넥톤         18
비에고         17
아펠리오스       16
루시안         14
dtype: int64

In [125]:
cahm_all_v14.value_counts().head(10)

champion
레오나         24
레넥톤         22
리 신         19
바루스         19
브라움         18
라이즈         17
그웬          16
아펠리오스       16
비에고         15
칼리스타        13
dtype: int64

In [126]:
cahm_all_v15.value_counts().head(10)

champion
리 신         28
레오나         21
비에고         17
아펠리오스       17
바루스         15
트런들         15
라이즈         14
나르          13
제이스         13
이즈리얼        12
dtype: int64

# 리그별 오브젝트 영향도 확인하기

In [3]:
LCK_object = LCK[['blue_object','blue_object_count','red_object','red_object_count']]
LCK_object

blue_object  blue_object_count      red_object  red_object_count
0         turret@blue                 11      turret@red                 1
1      inhibitor@blue                  3   inhibitor@red                 0
2          baron@blue                  1       baron@red                 0
3         dragon@blue                  4      dragon@red                 2
4     riftHerald@blue                  2  riftHerald@red                 0
...               ...                ...             ...               ...
1210      turret@blue                 11      turret@red                 2
1211   inhibitor@blue                  4   inhibitor@red                 0
1212       baron@blue                  2       baron@red                 0
1213      dragon@blue                  4      dragon@red                 2
1214  riftHerald@blue                  2  riftHerald@red                 0

[1215 rows x 4 columns]

In [4]:
a = LCK_object[LCK_object['blue_object'] == 'turret@blue']
aa = LCK_object[LCK_object['blue_object'] == 'inhibitor@blue']

In [5]:
LCK_turret = a.drop(columns=['blue_object','red_object'])
LCK_inhibitor = aa.drop(columns=['blue_object','red_object'])

In [6]:
LCK_turret.sum()

blue_object_count    1504
red_object_count     1398
dtype: int64

In [7]:
LCK_inhibitor.sum()

blue_object_count    233
red_object_count     210
dtype: int64

<pre>
서머시즌
LCK 는 블루팀일때 1504개, 레드팀일때 1398개
서머시즌 경기수는 226 경기로 + 선발전 12 + 플레이오프 18
블루팀은 경기당 5.8개
레드팀은 경기당 5.4개

억제기는 블루가 233개
레드가 210개로

경기당
블루팀  0.9개
레드팀 0.8개

경기당 11.2개의 포탑을 밀어내며
블루팀이 포탑을 밀어내는 운영에서 조금 더 이점을 가짐

</pre>

In [41]:
b = LCK_object[LCK_object['blue_object'] == 'riftHerald@blue']

In [42]:
LCK_riftHerald = b.drop(columns=['blue_object','red_object'])

In [43]:
LCK_riftHerald.sum()

blue_object_count    309
red_object_count     167
dtype: int64

전령 획득률이 블루팀이 309, 레드팀이 167

블루팀이 두배에 가까운 전령획득 능력을 가지는 것을 알 수 있음

이렇게 획득된 전령을 바탕으로 포탑에 대한 철거능력이 더 강했음을 알 수 있음

경기당 블루는 1.2마리
레드팀은 0.6마리의 전령을 챙겼음

블루팀이 일방적으로 2개를 가져가는 경우가 가장많으며

그 다음이 1개씩 나눠먹게되는 경우임

In [44]:
c = LCK_object[LCK_object['blue_object'] == 'dragon@blue']
LCK_dragon= c.drop(columns=['blue_object','red_object'])

In [45]:
LCK_dragon.sum()

blue_object_count    500
red_object_count     637
dtype: int64

반대로 드래곤의 경우 블루팀이 500개,  레드팀은 637개로

레드팀이 우세하게 가져가는 모습을 볼 수 있음

LCK의 팀들은 블루진영일 경우 전령에 대한 획득능력이 높으며 레드팀은 이에 대한 대응으로 드래곤을 획득하지 않을까 생각함

지리적인 요소가 블루팀의 전령획득률을 도와주는 것이 아닌가

경기당 드래곤 처치 수는 5마리

블루팀은 2.2마리

레드팀은 2.8마리를 처치함

In [13]:
d = LCK_object[LCK_object['blue_object'] == 'baron@blue']
LCK_baron= d.drop(columns=['blue_object','red_object'])

In [14]:
LCK_baron.sum()

blue_object_count    179
red_object_count     163
dtype: int64

In [11]:
(179+163)/256

1.3359375

In [10]:
163/256

0.63671875

In [50]:
LCK_baron.value_counts()

blue_object_count  red_object_count
0                  1                   69
1                  0                   69
2                  0                   27
0                  2                   23
1                  1                   23
0                  0                   15
1                  2                    6
2                  1                    3
3                  0                    3
2                  2                    2
1                  3                    1
3                  3                    1
4                  0                    1
dtype: int64

바론의 경우 큰 숫자의 차이는 없지만

블루팀 179, 레드팀 163으로 역시 블루팀이 우세함

# 왜 블루팀이 전령 획득률이 높은걸까?
## 첫 번째. 블루팀의 지형적 이점에 있다고 생각함
## 두 번째. 블루팀이 가지는 픽의 이점에서 발생한다고 생각함

In [15]:
LCK['blue_ban'].value_counts().head(5)

루시안     67
칼리스타    66
아칼리     51
제이스     50
바루스     49
Name: blue_ban, dtype: int64

In [18]:
LCK['red_ban'].value_counts().head(5)

레넥톤     94
루시안     85
신 짜오    67
리 신     50
제이스     50
Name: red_ban, dtype: int64

In [14]:
a =LCK['blue_champion'].value_counts().head(5)
a

리 신    70
비에고    53
레오나    53
레넥톤    52
럼블     51
Name: blue_champion, dtype: int64

In [15]:
b = LCK['red_champion'].value_counts().head(5)
b

레오나     68
리 신     57
다이애나    51
바루스     51
비에고     47
Name: red_champion, dtype: int64

블루팀과 레드팀 각각 가장 많이 픽된 챔피언을 확인해보았음

서머시즌 가장 많이 등장했던 리신,비에고,레오나 등의 픽이 확인됨

블루팀의 상위 5픽은 중반교전에 매우 강한 픽들임을 알 수 있음

레드팀의 상위 5픽에는 바텀에서의 픽이 두개나 포함되어있음


전령 등장 시간대는 8분으로 서포터가 6렙이 찍어질랑 말랑하는 타이밍임
정글러들은 궁극기를 갖추는 레벨임

바텀의 영향력이 밀릴 가능성이 높음 

블루팀의 상위 5픽에서 리신, 럼블, 레넥톤과 같은 픽이 

레드의 픽들에 비해 전령 싸움을 할 때 더욱 강한 픽이기에 전령싸움에서 이점을 가지는 것 아닐까

전령타임 더 레벨이 높은 미드,정글,탑의 영향력이 바텀에 비해 강하기에

이러한 픽적인 이점에서 블루쪽이 전령을더 많이 가져갈 수 있는 거 아닐까

<pre>
리신의 경우 블루팀이 13회
럼블은 21회
레넥톤은 30회 가까이 더 많이 픽되었음
</pre>

# 얻은 전령의 효과로 스노우볼

## 8분시간대 이후에 블루가 얼마나 많이굴릴수가있는가

블루가 전령

LCK의 바론의 경우 블루가 더 많긴한데

서로 한개씩 먹는 경우가 가장 많았고 횟수도 같음

그뒤로 한쪽이 2개를 일반적으로 먹는경우도 블루가 4회정도 많음

1:1은 23회로 다섯번째에 위치함

거의 바론은 서로 비슷하게 먹는다는 느낌임

그래도 블루팀이 유리한 부분이 있으니까 바론을 더 많이 먹긴했을거임

In [102]:
LCK_gold = pd.read_csv('projectData/lck_totalgold.csv', index_col=0)


In [103]:
LCK_gold = LCK_gold.reset_index()

In [104]:
gold=[]

In [105]:
for i in LCK_gold.index:
    if (i % 2) == 0:
        gold.append(i)

In [106]:
lckgold2 = LCK_gold.iloc[gold]
lckgold2

index    0    1    2     3     4     5     6     7     8  ...  59  60  61  \
0     BRO   77   22 -120   -18   244   792   761  1167  1257  ... NaN NaN NaN   
2     LSB   44  380  563   744  1059   726   498  1311   672  ... NaN NaN NaN   
4      T1   77 -228   23   720   700   868   682   918   814  ... NaN NaN NaN   
6     HLE  -41  280  -51  -800 -1249 -1367 -2056 -3025 -3197  ... NaN NaN NaN   
8      KT   36 -227  -46  1809  1529  1879  2023  2723  2973  ... NaN NaN NaN   
..    ...  ...  ...  ...   ...   ...   ...   ...   ...   ...  ...  ..  ..  ..   
478   GEN   -9    6 -413  -212  -109  -324  -909 -1300 -1497  ... NaN NaN NaN   
480    DK -104  888  949   949   965  1409  1471  1202   913  ... NaN NaN NaN   
482    DK  -61 -368 -376  -178  -105     9  -349   -81  -451  ... NaN NaN NaN   
484    T1  121  536  346   466   208   783   681  1780  2136  ... NaN NaN NaN   
486    T1  -22 -416 -509    24   139  -115   145   179   164  ... NaN NaN NaN   

     62  63  64  65  66  67  68  
0   NaN NaN NaN NaN NaN NaN NaN  
2   NaN NaN NaN NaN NaN NaN NaN  
4   NaN NaN NaN NaN NaN NaN NaN  
6   NaN NaN NaN NaN NaN NaN NaN  
8   NaN NaN NaN NaN NaN NaN NaN  
..   ..  ..  ..  ..  ..  ..  ..  
478 NaN NaN NaN NaN NaN NaN NaN  
480 NaN NaN NaN NaN NaN NaN NaN  
482 NaN NaN NaN NaN NaN NaN NaN  
484 NaN NaN NaN NaN NaN NaN NaN  
486 NaN NaN NaN NaN NaN NaN NaN  

[244 rows x 70 columns]

In [94]:
for i in range(69):
    lckgold2[f'{i}'] = abs(lckgold2[f'{i}'])

<ipython-input-94-94a118eb1f41>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  lckgold2[f'{i}'] = abs(lckgold2[f'{i}'])


In [101]:
lckgold2.set_index('index')[['6','7','8','9']]

6     731.434426
7     891.831967
8    1003.422131
9    1212.295082
dtype: float64

In [108]:
lckgold2[['6','7','8','9']].mean()

6     47.836066
7     15.438525
8    129.684426
9    169.811475
dtype: float64